api/v1alpha1/clusterextension_types.go

```diff
@@ -83,6 +83,19 @@ type ClusterExtensionSpec struct {
 	//+kubebuilder:Optional
 	// Preflight defines the configuration of preflight checks.
 	Preflight *PreflightConfig `json:"preflight,omitempty"`
+
+	// ServiceAccount is used to install and manage resources.
+	// The service account is expected to exist in the InstallNamespace.
+	ServiceAccount ServiceAccountReference `json:"serviceAccount"`
```

Comment:
Since we are attempting to follow the Kubernetes/OpenShift API conventions from previous comments by @joelanford, I think we should follow the naming convention for references which is `{field}Ref` as outlined by https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-of-the-reference-field

```suggestion
	ServiceAccountRef ServiceAccountReference `json:"serviceAccountRef"`
```

Reply:
Ah, I do see in the openshift conventions they state that the `Ref` suffix should be omitted. @joelanford Do we have any preference in the API conventions we want to lean towards?

Reply:
This is the one I followed https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#use-specific-types-for-object-references-and-omit-ref-suffix

Reply:
As mentioned in https://kubernetes.slack.com/archives/C0181L6JYQ2/p1720534180890229 I won't hold the PR on this, but we should begin to think about the API conventions we want to favor and highlight when and why we deviate.